### PR TITLE
Update vector repository and lint project

### PR DESCRIPTION
**Pull request by beechesII**
* Refactor Role (timber.io moved to datadog)
* Lint Project with actual linting rules
* add molecule tests for various systems
* move collection to namespace telekom_mms

squash on merge ;)

### CODE REVIEW

File: roles/vector/tasks/install.yml [hunk 1]
```diff
@@ -1,46 +1,31 @@
...
 
-    - name: Set latest version
-      set_fact:
-        vector_version: "{{ bucket_content.content | regex_replace('.*\\n.*<Key>[^-]+-(?P<version>[\\d\\.]+)-.*','\\g<version>') }}"
-      when: vector_version == "latest"
-
-    - name: Install Vector (Debian)
-      apt:
-        deb: "https://packages.timber.io/vector/{{ version }}/vector-{{ version }}-{{ arch }}.deb"
+- name: Install vector from source
```
Comment (z-bsod) [inline roles/vector/tasks/install.yml]: with "install from source" i'd expect a compilation step here. I guess "install vector directly from package" would be a better description for what is happening here

File: .github/workflows/new_version.yml [hunk 2]
```diff
@@ -1,24 +0,0 @@
----
```
Comment (schurzi) [inline .github/workflows/new_version.yml:1]: This workflow was deleted without a substitute?

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -1,33 +0,0 @@
----
-name: Tests
-
-on: 
-  push:
-  pull_request:
-  schedule:
-    - cron: '20 10 * * tue'
```
Comment (schurzi) [inline .github/workflows/push.yml:8]: the actions for `push` and `schedule` are missing from the new workflow

File: .github/workflows/molecule.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          - ubuntu2204
+          - ubuntu2404
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: "${{ github.repository }}"
+
+      - name: Molecule
+        uses: gofrolist/molecule-action@v2
```
Comment (schurzi) [inline .github/workflows/molecule.yml]: nice, didn't know this, good find!

File: roles/vector/molecule/debian12/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (schurzi) [inline roles/vector/molecule/debian12/molecule.yml]: why are the Molecule tests inside the role?
I usually place then in the collection root and outside any implementation. Is there s specific reason for doing it this way?

File: roles/vector/tasks/repo.yml [hunk 6]
```diff
@@ -1,25 +1,60 @@
 ---
 
-- block:
-    - name: Add vector gpg key to apt
-      become: true
-      apt_key:
-        url: "{{ vector_repo_key }}"
+- name: Fetch OS dependent variables
```
Comment (schurzi) [inline roles/vector/tasks/repo.yml]: I see no OS specific vars, is thin needed somewhere?

**Review by schurzi — COMMENTED** [review]
Why is there an empty `CHANGELOG.md`?

File: roles/vector/tasks/install.yml (hunk 1 above)
Comment (beechesII) [reply roles/vector/tasks/install.yml]: changed :)

File: roles/vector/molecule/debian12/molecule.yml (hunk 5 above)
Comment (beechesII) [reply roles/vector/molecule/debian12/molecule.yml]: Referring to the site https://www.ansible.com/blog/developing-and-testing-ansible-roles-with-molecule-and-podman-part-1/ it seems that the correct place for molecule directory is inside the role directory. On this site https://ansible.readthedocs.io/projects/molecule/getting-started/ the molecule folder is outsite the roles path. I think there is no best practise.

File: .github/workflows/new_version.yml (hunk 2 above)
Comment (beechesII) [reply .github/workflows/new_version.yml:1]: This worklfow was from the original repository. I think we can a a simular workflow after reference our repository in the original one.

File: roles/vector/tasks/repo.yml (hunk 6 above)
Comment (beechesII) [reply roles/vector/tasks/repo.yml]: Your right. I removed this task.

**Comment by beechesII** [comment]
> Why is there an empty `CHANGELOG.md`?

I added this as a new file.

File: .github/workflows/push.yml (hunk 3 above)
Comment (beechesII) [reply .github/workflows/push.yml:8]: Added them back to molecule.yml and ansible-lint.yml.

File: roles/vector/molecule/debian12/molecule.yml (hunk 5 above)
Comment (schurzi) [reply roles/vector/molecule/debian12/molecule.yml]: hmm, I'd really like to move the molecule files outside the role. It seems way cleaner to me.

Comment (beechesII) [reply roles/vector/molecule/debian12/molecule.yml]: Done with Commit https://github.com/telekom-mms/ansible-vector/pull/2/commits/2f51d0841ea6563c72c7b29eec8c456adfbdf982.

File: molecule/ubuntu2404/molecule.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+---
+platforms:
+  - name: ubuntu2404
+    image: geerlingguy/docker-ubuntu2404-ansible:latest
```
Comment (schurzi) [inline molecule/ubuntu2404/molecule.yml]: molecule supports variables here, so we yould unify all tests to one scenario.

File: molecule/ubuntu2204/verify.yml [hunk 8]
```diff
@@ -0,0 +1 @@
+../default/verify.yml
```
Comment (schurzi) [inline molecule/ubuntu2204/verify.yml]: this works? ... didn't know that solution, nice!

**Review by schurzi — APPROVED** [review]

File: roles/vector/molecule/debian12/molecule.yml (hunk 5 above)
Comment (beechesII) [reply roles/vector/molecule/debian12/molecule.yml]: changed, I need to fix rockylinux8 test

File: molecule/ubuntu2404/molecule.yml (hunk 7 above)
Comment (beechesII) [reply molecule/ubuntu2404/molecule.yml]: implemented variable support

**Pull request merged**